gson/src/main/java/com/google/gson/internal/LinkedTreeMap.java

```diff
@@ -69,12 +81,15 @@ public LinkedTreeMap() {
    *
    * @param comparator the comparator to order elements with, or {@code null} to
    *     use the natural ordering.
+   * @param allowNullValues whether {@code null} is allowed as entry value
    */
   @SuppressWarnings({ "unchecked", "rawtypes" }) // unsafe! if comparator is null, this assumes K is comparable
-  public LinkedTreeMap(Comparator<? super K> comparator) {
+  public LinkedTreeMap(Comparator<? super K> comparator, boolean allowNullValues) {
```

Comment:
This is an incompatible change, for client code that might be calling `new LinkedTreeMap<>(comparator)` directly. Of course it's an `.internal.` class, so code shouldn't be doing that. I find that Google's codebase has a few places where people _are_ constructing `LinkedTreeMap`, but none where they are calling the `(Comparator)` constructor. So I think we could actually just delete that constructor and hardcode the use of natural ordering. That would allow considerable simplification of the `find` method.

Reply:
It looks like other projects [are using `LinkedTreeMap` as well](https://sourcegraph.com/search?q=context:global+new+LinkedTreeMap&patternType=standard), though they don't seem to use that constructor either. Should I add it back nonetheless?

> So I think we could actually just delete that constructor and hardcode the use of natural ordering. That would allow considerable simplification of the `find` method.

Sounds in general reasonable to me, but I am not sure whether that should be part of this pull request.

Reply:
OK, we can do that as a follow-on.